src/transactions/saga.ts

```diff
@@ -307,3 +313,58 @@ export function* transactionSaga() {
   yield* spawn(watchAddressToE164PhoneNumberUpdate)
   yield* spawn(watchPendingTransactions)
 }
+
+export function* handleTransactionReceiptReceived(
+  txId: string,
+  receipt: TransactionReceipt | CeloTxReceipt,
+  networkId: NetworkId
+) {
+  const tokensById = yield* select((state) => tokensByIdSelector(state, [networkId]))
+
+  // TODO: Receive feeCurrencyId from transaction request.
+  const feeCurrencyId = getTokenId(networkId)
+  const feeTokenInfo = tokensById[feeCurrencyId]
+
+  if (!feeTokenInfo) {
+    Logger.warn(`No information found for token ${feeCurrencyId} in network ${networkId}`)
+  }
+
+  const gasFeeInWei = new BigNumber(receipt.gasUsed.toString()).times(
+    new BigNumber(receipt.effectiveGasPrice.toString())
+  )
+
+  const baseDetails = {
+    transactionHash: receipt.transactionHash,
+    block: receipt.blockNumber.toString(),
+    status:
+      receipt.status === 'reverted' || !receipt.status
+        ? TransactionStatus.Failed
+        : TransactionStatus.Complete,
+  }
+
+  // Only add fee data in non-celo transactions
+  const feeDetails =
+    !feeTokenInfo || networkId === networkConfig.defaultNetworkId
+      ? {}
+      : {
+          fees: buildGasFees(feeTokenInfo, gasFeeInWei),
```

Comment:
nit: any reason why you decided to create this object and method? why not have the condition directly for the `fees` property of the action payload, where the value would be an empty array or an array with the security fee? e.g. 
```
transactionConfirmed(txId, {
      ...baseDetails,
      fees: !feeTokenInfo || networkId === networkConfig.defaultNetworkId ? [] : [...]
    })
```
?

Reply:
Nop, let me clean this. I think I'm gonna leave the function because I think it's easier to read.